src/iden3comm/handlers/auth.ts

```diff
@@ -158,19 +156,14 @@ export class AuthHandler implements IAuthHandler {
     };
 
     for (const proofReq of authRequest.body.scope) {
-      const zkpReq: ZeroKnowledgeProofRequest = {
-        id: proofReq.id,
-        circuitId: proofReq.circuitId as CircuitId,
-        query: proofReq.query
-      };
-
       const query = proofReq.query as unknown as ProofQuery;
 
       const zkpRes: ZeroKnowledgeProofResponse = await this._proofService.generateProof(
-        zkpReq,
+        proofReq,
         did,
         {
-          skipRevocation: query.skipClaimRevocationCheck ?? false
+          skipRevocation: query.skipClaimRevocationCheck ?? false,
+          verifierDid: DID.parse(authRequest.from)
```

Comment:
I guess we need to add group id logic here ? because we should generate linkNonce based on group id even for v3 circuit.